src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/StatDataService.cs

```diff
@@ -0,0 +1,170 @@
+            VeiledPatterns = new List<StatData>();
+
+            NewLinePattern = new Regex("(?:\\\\)*[\\r\\n]+");
+
+            var hashPattern = new Regex("\\\\#");
+            var parenthesesPattern = new Regex("((?:\\\\\\ )*\\\\\\([^\\(\\)]*\\\\\\))");
+
+            foreach (var category in categories)
+            {
+                var first = category.Entries.FirstOrDefault();
+                if (first == null)
+                {
+                    continue;
+                }
+
+                // The notes in parentheses are never translated by the game.
+                // We should be fine hardcoding them this way.
+                string suffix, pattern;
+                List<StatData> patterns;
+                switch (first.Id.Split('.').First())
+                {
+                    default: continue;
+                    case "delve":
+                    case "monster":
+                    case "explicit": suffix = "\\ *\\n+"; patterns = ExplicitPatterns; break;
+                    case "implicit": suffix = "\\ *\\(implicit\\)"; patterns = ImplicitPatterns; break;
+                    case "enchant": suffix = "\\ *\\(enchant\\)"; patterns = EnchantPatterns; break;
+                    case "crafted": suffix = "\\ *\\(crafted\\)"; patterns = CraftedPatterns; break;
+                    case "veiled": suffix = "\\ *\\(veiled\\)"; patterns = VeiledPatterns; break;
+                }
+
+                foreach (var entry in category.Entries)
+                {
+                    entry.Category = category.Label;
+
+                    pattern = Regex.Escape(entry.Text);
+                    pattern = parenthesesPattern.Replace(pattern, "(?:$1)?");
+                    pattern = hashPattern.Replace(pattern, "([-+\\d,\\.]+)");
+                    pattern = NewLinePattern.Replace(pattern, "\\n");
+
+                    entry.Pattern = new Regex($"\\n+{pattern}{suffix}");
+                    patterns.Add(entry);
+                }
+            }
+        }
+
+        public Mods ParseMods(string text)
+        {
+            text = NewLinePattern.Replace(text, "\n");
+
+            var mods = new Mods();
+
+            // Make sure the text ends with an empty line for our regexes to work correctly
+            if (!text.EndsWith("\n"))
+            {
+                text += "\n";
+            }
+
+            FillMods(mods.Explicit, ExplicitPatterns, text);
+            FillMods(mods.Implicit, ImplicitPatterns, text);
+            FillMods(mods.Enchant, EnchantPatterns, text);
+            FillMods(mods.Crafted, CraftedPatterns, text);
+            // FillMods(mods.Veiled, VeiledPatterns, text);
+
+            return mods;
+        }
+
+        private void FillMods(List<Mod> mods, List<StatData> patterns, string text)
+        {
+            var results = patterns
+                .Where(x => x.Pattern.IsMatch(text))
+                .ToList();
+
+            foreach (var x in results)
```

Comment:
Shouldn't do `.ToList()` here, it will result in iterating through the results twice.